src/System.Runtime/tests/System/Text/StringBuilderTests.cs

```diff
@@ -1436,12 +1473,30 @@ public static void Remove_Invalid()
             var builder = new StringBuilder("Hello");
             Assert.Throws<ArgumentOutOfRangeException>("startIndex", () => builder.Remove(-1, 0)); // Start index < 0
             Assert.Throws<ArgumentOutOfRangeException>("length", () => builder.Remove(0, -1)); // Length < 0
+        }
+
+        [Fact]
+        [SkipOnTargetFramework(~TargetFrameworkMonikers.NetFramework, "Parameter name changes in error messages are breaking, not currently backported.")]
+        public static void Remove_Index_Invalid()
+        {
+            var builder = new StringBuilder("Hello");
 
             Assert.Throws<ArgumentOutOfRangeException>("index", () => builder.Remove(6, 0)); // Start index + length > 0
```

Comment:
There is a way to keep a single test and dynamically change the behavior according to the platform, that would allow you to coalesce several tests here. I"ll see if I can find out how.